corehq/apps/reports/standard/cases/case_list_explorer.py

```diff
@@ -68,6 +73,9 @@ def _build_query(self, sort=True):
         if xpath:
             try:
                 query = query.xpath_query(self.domain, xpath)
+                node = parse_xpath(xpath)
+                if is_ancestor_comparison(node) or 'ancestor-exists' in xpath:
+                    validate_ancestor_query_limit(query, node)
```

Comment:
Can you elaborate on the goal of this limit?  My read of this is it limits the results returned in the CLE to 500k when there's an ancestor query in the xpath, and not otherwise.  However, the size of the results is already limited by the page size in the ancestor class here:
https://github.com/dimagi/commcare-hq/blob/69cff2e739a90f3fc5ee6a10cda7cbab1b9fa017/corehq/apps/reports/standard/cases/basic.py#L51-L51
I think the exports version doesn't have a limit applied (though it looks like the code above would cap it at 10??  I haven't tried it out):
https://github.com/dimagi/commcare-hq/blob/69cff2e739a90f3fc5ee6a10cda7cbab1b9fa017/corehq/apps/reports/standard/cases/case_list_explorer.py#L164-L164
Either way, what's the rationale for capping the export size of the CLE just for ancestor queries, if the ancestor part already passed size validation?

Reply:
Oh right - `size` has a different meaning in the context of scroll queries, that explains why exports aren't limited:
https://github.com/dimagi/commcare-hq/blob/69cff2e739a90f3fc5ee6a10cda7cbab1b9fa017/corehq/apps/es/es_query.py#L360-L367

Reply:
hey @esoergel! Yea that's a valid point. Originally I wanted to mirror the limiting that was happening with the subcase queries but I guess that serves the same purpose as the size validation that's happening in the ancestor queries. My rationale now is that technically, it's still possible to ancestor queries to return more than 500k cases so I'm placing a (probably overkill) extra limit at the end. An example scenario could be like if the search term `parent/age > 45`, returns 400k cases (matches the `age > 45` part), and of those, 300k have 2 child cases each, eventually returning 600k cases at the end. It's kinda arbitrary at this point but if you still have concerns I'd be fine with closing this PR too.

Reply:
> it's still possible to ancestor queries to return more than 500k cases

I guess what I'm getting at is that it's also possible for _non_ ancestor queries to return more than 500k cases, and I don't follow why we'd want a limit on the one but not the other.  I do think it's superfluous to do the validation here, as there are limits applied in other places depending on how the query is called, and I worry that it adds complexity unnecessarily.

Reply:
sorry again about the super delayed responses here - I keep getting pulled away by something different but yea I do agree that this doesn't really add anything. I'll close this out.